drivers/docker/util/util_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package util
+
+import (
+	"testing"
+)
+
+func TestCalculateCPUPercent(t *testing.T) {
+	tests := []struct {
+		name      string
+		newSample uint64
+		oldSample uint64
+		newTotal  uint64
+		oldTotal  uint64
+		cores     int
+		expected  float64
+	}{
+		{"valid case", 100, 50, 200, 100, 4, 200.0},
+		{"zero denominator", 100, 50, 100, 100, 4, 0.0},
+		{"negative case", 50, 100, 100, 200, 4, 0.0},
```

Comment:
nit: using positional initialisation rather than named makes it a little harder to read and understand what is going on.